src/transmission_manager.rs

```diff
@@ -0,0 +1,341 @@
+            InfectiousStatusType::Recovered,
+        );
+    } else {
+        // Schedule the next infection attempt.
+        // Get the next infection attempt time.
+        let next_infection_times = get_next_infection_time(context, last_infection_time_uniform);
+
+        // Schedule the infection attempt. This function (a) grabs a contact at the time of the infection event
+        // to ensure that the contacts are current and (b) evaluates whether the transmission event is successful.
+        context.add_plan(
+            context.get_current_time() + next_infection_times.gi,
+            move |context| {
+                infection_attempt(context, transmitter_id)
+                    .expect("Error finding contact in infection attempt");
+                // Schedule the next infection attempt for this infected agent
+                // once the last infection attempt is over.
+                schedule_next_infection_attempt(
+                    context,
+                    transmitter_id,
+                    num_infection_attempts_remaining - 1,
+                    next_infection_times.uniform,
+                );
+            },
+        );
+    }
+}
+
+/// Calculate the next infection time. Draws an increasing value of a uniform distribution
+/// and passes it through the inverse CDF of the generation interval to get the next infection time.
+fn get_next_infection_time(
+    context: &mut Context,
+    last_infection_time_uniform: f64,
+) -> InfectionTimes {
+    // FOR NOW, we are using placeholder math to guarantee the infection times are always increasing.
+    // This is not order statistics. This will be corrected at a later time.
+    let next_infection_time_unif =
+        context.sample_range(TransmissionRng, last_infection_time_uniform..1.0);
+
+    // In the future, we will generalize the use of an exponential distribution to
+    // an arbitrary distribution with a defined inverse CDF.
+    let gi = context
+        .get_global_property_value(Parameters)
+        .unwrap()
+        .generation_interval;
+    let next_infection_time_gi = Exp::new(1.0 / gi)
+        .unwrap()
+        .inverse_cdf(next_infection_time_unif);
+
+    InfectionTimes {
+        uniform: next_infection_time_unif,
+        gi: next_infection_time_gi,
+    }
+}
+
+/// This function is called when an infected agent has an infection event scheduled.
+/// This function includes identifying a potential contact to infect, evaluating whether
+/// the transmission event is successful, and scheduling the next infection event.
+fn infection_attempt(context: &mut Context, transmitter_id: PersonId) -> Result<(), IxaError> {
+    // This is a method from a trait extension implemented in `mod contact`.
+    // As long as the method returns a contact id, it can use any underlying sampling strategy
+    // to obtain that contact, and that strategy can be separately implemented in `mod contact`
+    // without changing the logic here.
```

Comment:
I'm not sure I understand this comment. I tend to think that the note about there being no guarantee about the strategy should go with the definition.

Per our previous discussion, do we want to have this be a trait extension or a function.

Reply:
> Per our previous discussion, do we want to have this be a trait extension or a function.

In this case, I'm totally agnostic. It would be a private trait, right? In that case, I don't see any difference to what's implemented right now, and I am happy with whatever you or @k88hudson-cfa think is the best style-wise.